### PR TITLE
Feat/add randomness tests

### PR DESCRIPTION
**Pull request by JordyRo1**

### CODE REVIEW

**Comment by JordyRo1** [comment]
Issue while working on the katana integration with the randomness tests: 
We use for testing purpose on katana the same `FEE_TOKEN` that for the real testnet (Starkgate ETH). Since we are working on a Katana fork, the balance will not be provided with an initial amount as for the initial devnet. This lead to OUT_OF_GAS error for every test, since the balance for that ETH token will always be null. There is a solution, which is instead, for testing purpose, deploy an additional contract on the `declare_deploy_randomness`: an erc20, where we will mint an initial amount for the `account.address` (which is linked to the `DEVNET_PRE_DEPLOYED_ACCOUNT `) thus providing an initial balance. We will have to provide the sierra and casm of an erc20 for this to work.

**Comment by JordyRo1** [comment]
Notes: in the documentation and the sdk, a clarification needs to be made. Minimum block number and publish delay are two different notions. Minimum_block_number = current_block_number + publish_delay. When we first initiate the request_random, we provide the publish delay and the contract fetch the current block number. This information is later used in the hashing process. When the user wants to cancel a request / the submit random process is initiate, we need to have the minimum_block_number as a parameter. So we need to block number associated to the request_random, not the one associated to the submit_random/cancel_random.

**Comment by JordyRo1** [comment]
Should be all good now. I deployed an erc20 on the tests in order to centralize every elements of the test. There is one test failing, because of 'request already fulfilled'. This is an expected event.

**Comment by EvolveArt** [comment]
@JordyRo1 As far as I know, the integration tests are not running on a Katana Fork. Only the `upgrade` one is. So no need to bother with that.

> Issue while working on the katana integration with the randomness tests: We use for testing purpose on katana the same `FEE_TOKEN` that for the real testnet (Starkgate ETH). Since we are working on a Katana fork, the balance will not be provided with an initial amount as for the initial devnet. This lead to OUT_OF_GAS error for every test, since the balance for that ETH token will always be null. There is a solution, which is instead, for testing purpose, deploy an additional contract on the `declare_deploy_randomness`: an erc20, where we will mint an initial amount for the `account.address` (which is linked to the `DEVNET_PRE_DEPLOYED_ACCOUNT `) thus providing an initial balance. We will have to provide the sierra and casm of an erc20 for this to work.

File: pragma/core/mixins/randomness.py [hunk 1]
```diff
@@ -256,3 +394,24 @@ async def handle_random(self, private_key: int, min_block: int = 0):
 
                 # Wait for Tx to pass
                 await asyncio.sleep(5)
+
+    async def get_last_example_random(self):
+        (response,) = await self.example_randomness.functions["get_last_random"].call()
+        return response
+
+    async def example_request_random(
```
Comment (EvolveArt) [inline pragma/core/mixins/randomness.py:402]: let's not add example contract related functions in the SDK. It's an external contract.

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**